modules/course_management/cog.py

```diff
@@ -77,23 +86,97 @@ async def add_course(
 		guild_ids=[config.guild_id],
 		options=[
 			create_option(
-				name="channel",
-				description="Mention a course the professor teaches. (eg. #automata)",
+				name="course",
+				description=(
+					"The channel corresponding to the course you want to delete."
+				),
 				option_type=SlashCommandOptionType.CHANNEL,
 				required=True,
 			),
 		],
+		connector={"course": "channel"},
 	)
 	@has_permissions(manage_channels=True)
 	async def delete_course(self, ctx: SlashContext, channel: discord.TextChannel):
-		await course_deleter.delete_course(ctx, channel.id)
+		await ctx.defer()
+		await course_deleter.delete_course(ctx, channel)
 		await ctx.send(
 			embed=embed_success(
 				"Well done... All evidence of that course has been deleted from the"
 				" face of the earth. Hope you're proud of yourself."
 			)
 		)
 
+	@cog_ext.cog_subcommand(
+		base="course",
+		name="activate",
+		description="Move an inactive course channel to the active courses list.",
+		guild_ids=[config.guild_id],
+		options=[
+			create_option(
+				name="course",
+				description=(
+					"The channel corresponding to the course you want to activate."
+				),
+				option_type=SlashCommandOptionType.CHANNEL,
+				required=True,
+			)
+		],
+	)
+	async def activate_course(self, ctx: SlashContext, course: discord.TextChannel):
```

Comment:
Is this meant to not require manage_channels permission? I think it's fine that way, just checking.

Reply:
Correct